src/store/transactions/requests.ts

```diff
@@ -28,19 +30,26 @@ export const requestTokenState = atom<string | null>({
 });
 
 export const requestTokenPayloadState = selector({
-  key: KEYS.REQUEST_TOKEN_PAYLOAD,
+  key: RequestKeys.REQUEST_TOKEN_PAYLOAD,
   get: ({ get }) => {
     const token = get(requestTokenState);
     return token ? getPayloadFromToken(token) : null;
   },
 });
 
 export const transactionRequestStxAddressState = selector({
-  key: KEYS.ADDRESS,
+  key: RequestKeys.ADDRESS,
   get: ({ get }) => get(requestTokenPayloadState)?.stxAddress,
 });
 
 export const transactionRequestNetwork = selector({
-  key: KEYS.NETWORK,
+  key: RequestKeys.NETWORK,
   get: ({ get }) => get(requestTokenPayloadState)?.network,
 });
+
+export const feeRateState = selector({
+  key: RequestKeys.FEE,
+  get: async ({ get }) => {
+    return get(feesApiClientState).getFeeTransfer() as unknown as Promise<number>;
+  },
+});
```

Comment:
I wonder if this should actually not go in this file. This file is meant more to hold any atoms related to the request token (jwt). This fee selector seems to be more like an api request.

Reply:
I've created an API req file containing this. 

Is there any reason to put the API clients in the store, versus exposing them via hooks (`useMemo` with a dep on network?).

Reply:
Yes because you cannot use hooks in atoms and often we need a statefully aware client in other atoms. It would fine to use these client atoms to create hooks, but not the other way around.

Reply:
> Is there any reason to put the API clients in the store, versus exposing them via hooks (useMemo with a dep on network?).

Also, we should ideally always have atoms be the place where fetching of data happens, and not directly in hooks (if that was what you were suggesting).

Reply:
Which code is responsible for cache invalidation/refetching etc, all the features of swr/react-query? Can we use any of these libraries or do we have to implement it ourselves ?

Reply:
> Which code is responsible for cache invalidation/refetching etc, all the features of swr/react-query? Can we use any of these libraries or do we have to implement it ourselves ?

Not sure what you're getting at, can you rephrase?

We can make use of react-query, but we should dig into that after the jotai migration.

Reply:
These libraries bring a lot of functionality, such as cache invalidation and methods to invoke refetching of data. 

We don't have these libraries, so wondering where, in this apps' architecture, where that functionality would reside.

Reply:
ah okay yes, makes sense! I 100% want to use react-query as I would hate to recreate and maintain all that functionality. The amazing thing about jotai is you can leverage all of the functionality of react-query but in atom state. [see here in heystack](https://github.com/blockstack/heystack/blob/main/src/store/hey.ts#L27) on how that's used. This is better than relying on using hooks for react-query because we can't then use the data they fetch in store without doing some higher level syncing of their fetches and putting it back into atom state (eg react-query hook -> onComplete -> setAtom(data)). With the use of atomWithQuery, we get all the stuff we love, and other atoms can rely directly on those query atoms and the data they fetch, and when they update, all the atoms that depend on it update too!